src/main/java/com/avioconsulting/mule/connector/vault/provider/internal/vault/client/auth/AWSIAMAuthenticator.java

```diff
@@ -45,10 +62,46 @@ public String getAuthPayload(VaultConfig config) throws VaultException {
             payload.addProperty("role", this.role);
         }
 
+        String encodedIamRequestHeaders = new String(Base64.getEncoder().encode(generateHeaders().getBytes(StandardCharsets.UTF_8)));
+
         payload.addProperty("iam_http_request_method", "POST");
         payload.addProperty("iam_request_url", this.iamRequestUrl);
-        payload.addProperty("iam_request_headers", this.iamRequestHeaders);
+        payload.addProperty("iam_request_headers", encodedIamRequestHeaders);
         payload.addProperty("iam_request_body", this.iamRequestBody);
         return payload.toString();
     }
+
+    private String generateHeaders() {
+
+        TreeMap<String, String> awsHeaders = new TreeMap();
+        awsHeaders.put("host", DEFAULT_HOST);
+        getIamServerId().ifPresent(serverId -> awsHeaders.put("x-vault-aws-iam-server-id", serverId));
+
+        AWSV4Auth awsV4Auth = new AWSV4Auth.Builder(iamAccessKey, iamSecretKey)
+                .regionName(DEFAULT_REGION)
+                .serviceName(DEFAULT_SERVICE_NAME)
+                .httpMethodName(DEFAULT_METHOD)
+                .canonicalURI(DEFAULT_CANONICAL_URI)
+                .queryParametes(null)
+                .awsHeaders(awsHeaders)
+                .payload(DEFAULT_PAYLOAD)
```

Comment:
If the user specifies an iamRequestBody, that's what we'll need to sign. Otherwise, we can use the default value.